packages/dapp/components/staking/BondingSharesExplorer.tsx

```diff
@@ -98,48 +103,59 @@ export const BondingSharesExplorerContainer = ({ managedContracts, web3Provider,
-          const isAllowed = await bondingToken.isApprovedForAll(walletAddress, bonding.address);
-          if (!isAllowed) {
-            // Allow bonding contract to control account share
-            if (!(await performTransaction(bondingToken.connect(signer).setApprovalForAll(bonding.address, true)))) {
-              return; // TODO: Show transaction errors to user
+          try {
+            const isAllowed = await bondingToken.isApprovedForAll(walletAddress, bonding.address);
+            if (!isAllowed) {
+              // Allow bonding contract to control account share
+              if (!(await performTransaction(bondingToken.connect(signer).setApprovalForAll(bonding.address, true)))) {
+                return; // TODO: Show transaction errors to user
+              }
             }
-          }
-
-          const bigNumberAmount = amount ? ethers.utils.parseEther(amount.toString()) : allLpAmount(id);
-          await performTransaction(bonding.connect(signer).removeLiquidity(bigNumberAmount, BigNumber.from(id)));
 
-          fetchSharesInformation();
-          refreshBalances();
+            const bigNumberAmount = amount ? ethers.utils.parseEther(amount.toString()) : allLpAmount(id);
+            await performTransaction(bonding.connect(signer).removeLiquidity(bigNumberAmount, BigNumber.from(id)));
+          } catch (error) {
+            console.log(`Withdrawing LP from ${id} failed:`, error);
+            // throws exception to update the transaction log
+            throw error;
+          } finally {
+            fetchSharesInformation(id);
+            refreshBalances();
+          }
         });
       },
       [model]
     ),
 
     onClaimUbq: useCallback(
       async (id) => {
-        if (!model || model.processing) return;
+        if (!model) return;
         console.log(`Claiming Ubiquity Governance token rewards from ${id}`);
-        setModel({ ...model, processing: true });
+        setModel((prevModel) => (prevModel ? { ...prevModel, processing: [...prevModel.processing, id] } : null));
         doTransaction("Claiming Ubiquity Governance tokens...", async () => {
-          await performTransaction(masterChef.connect(signer).getRewards(BigNumber.from(id)));
-
-          fetchSharesInformation();
-          refreshBalances();
+          try {
+            await performTransaction(masterChef.connect(signer).getRewards(BigNumber.from(id)));
+          } catch (error) {
+            console.log(`Claiming Ubiquity Governance token rewards from ${id} failed:`, error);
+            // throws exception to update the transaction log
+            throw error;
+          } finally {
+            fetchSharesInformation(id);
+            refreshBalances();
+          }
         });
       },
       [model]
     ),
 
     onStake: useCallback(
       async ({ amount, weeks }) => {
-        if (!model || model.processing) return;
+        if (!model || model.processing.length) return;
```

Comment:
Does stake have to be blocked while a claim/withdrawal occurs?

Reply:
you could stake while claiming rewards as it will not affect the rewards amount.
however Withdrawing LP tokens should disable staking LP tokens